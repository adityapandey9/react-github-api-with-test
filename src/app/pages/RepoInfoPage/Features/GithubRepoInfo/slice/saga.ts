import { call, put, select, takeLatest, delay } from 'redux-saga/effects';
import { request } from 'utils/request';
import { githubRepoInfoActions as actions } from '.';
import { FileContentRepo, Repo, RepoFileStruct } from 'types/Repo';
import { RepoErrorType } from './types';
import { selectUsername } from '../../../../HomePage/Features/GithubRepoForm/slice/selectors';
import {
  selectRepoListRepoId,
  selectRepoListRepos,
} from '../../../../ReposListPage/Features/GithubRepoList/slice/selectors';

/**
 * Github repos request/response handler
 */
export function* getRepoById() {
  yield delay(500);
  // Select username from store
  const username: string = yield select(selectUsername);
  if (username.length === 0) {
    yield put(actions.repoError(RepoErrorType.USERNAME_EMPTY));
    return;
  }
  const repoId: string = yield select(selectRepoListRepoId);
  if (repoId.length === 0) {
    yield put(actions.repoError(RepoErrorType.USERNAME_EMPTY));
    return;
  }

  try {
    let repos: Repo[] = yield select(selectRepoListRepos);

    if (repos.length === 0) {
      const requestURL = `https://api.github.com/users/${username}/repos?type=all&sort=updated`;
      repos = yield call(request, requestURL);
    }

    const filterRepo = repos.filter(item => item.name === repoId);

    if (
      filterRepo.length === 0 ||
      (filterRepo.length > 0 && !filterRepo[0].default_branch)
    ) {
      yield put(actions.repoError(RepoErrorType.USER_NOT_FOUND));
      return;
    }

    const repo: Repo = filterRepo[0];
    const requestURL = `https://api.github.com/repos/${username}/${repoId}/git/trees/${repo.default_branch}?recursive=1`;

    // Call our request helper (see 'utils/request')
    const repoInfo: RepoFileStruct = yield call(request, requestURL);
    if (!(repoInfo?.tree && repoInfo.tree.length > 0)) {
      yield put(actions.repoError(RepoErrorType.USER_HAS_NO_REPO));
      return;
    }

    const doesReadMDFileExist = repoInfo.tree.filter(item =>
      item.path.toLowerCase().includes('readme.md'),
    );

    repoInfo.tree = repoInfo.tree.filter(item => !item.path.includes('/'));

    if (doesReadMDFileExist.length > 0) {
      const readMeURL = doesReadMDFileExist[0].url;
      // Call our request helper (see 'utils/request')
      const readMeFile: FileContentRepo = yield call(request, readMeURL);
      repoInfo.readMe = atob(readMeFile.content);
    }

    yield put(actions.reposLoaded(repoInfo));
  } catch (err) {
    if (err.response?.status === 404) {
      yield put(actions.repoError(RepoErrorType.USER_NOT_FOUND));
    } else if (err.message === 'Failed to fetch') {
      yield put(actions.repoError(RepoErrorType.GITHUB_RATE_LIMIT));
    } else {
      yield put(actions.repoError(RepoErrorType.RESPONSE_ERROR));
    }
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export function* githubRepoInfoSaga() {
  // Watches for loadRepos actions and calls getRepos when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(actions.loadRepoById.type, getRepoById);
}
