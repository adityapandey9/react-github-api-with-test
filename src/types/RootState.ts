import { GithubRepoFormState } from 'app/pages/HomePage/Features/GithubRepoForm/slice/types';
import { ThemeState } from 'styles/theme/slice/types';
import { GithubRepoListState } from '../app/pages/ReposListPage/Features/GithubRepoList/slice/types';
import { GithubRepoInfoState } from '../app/pages/RepoInfoPage/Features/GithubRepoInfo/slice/types';
// [IMPORT NEW CONTAINER STATE ABOVE] < Needed for generating containers seamlessly

/* 
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
  Properties are optional because they are injected when the components are mounted sometime in your application's life. 
  So, not available always
*/
export interface RootState {
  theme?: ThemeState;
  githubRepoForm?: GithubRepoFormState;
  githubRepoList?: GithubRepoListState;
  githubRepoInfo?: GithubRepoInfoState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
