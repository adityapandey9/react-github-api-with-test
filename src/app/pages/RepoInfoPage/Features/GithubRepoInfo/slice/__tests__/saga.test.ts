import { put, takeLatest } from 'redux-saga/effects';
import * as slice from '..';

import { githubRepoInfoSaga, getRepoById } from '../saga';
import { RepoErrorType } from '../types';

describe('getRepoById Saga', () => {
  let username: any;
  let repoId: any;
  let repos: any;
  let getReposIterator: ReturnType<typeof getRepoById>;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getReposIterator = getRepoById();
    const delayDescriptor = getReposIterator.next().value;
    expect(delayDescriptor).toMatchSnapshot();

    const selectDescriptor = getReposIterator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should return error if username is empty', () => {
    username = '';
    let putDescriptor = getReposIterator.next(username).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USERNAME_EMPTY)),
    );

    const iteration = getReposIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('should return error if repoId is empty', () => {
    username = 'test_username';
    repoId = '';
    getReposIterator.next(username);
    let putDescriptor = getReposIterator.next(repoId).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USERNAME_EMPTY)),
    );

    const iteration = getReposIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('should return error if repos is empty', () => {
    username = 'test_username';
    repoId = 'test_repo';
    repos = [];
    getReposIterator.next(username);
    getReposIterator.next(repoId);
    getReposIterator.next(repos);
    let putDescriptor = getReposIterator.next(repos).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USER_NOT_FOUND)),
    );

    const iteration = getReposIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('should dispatch the reposLoaded action if it requests the data without default branch unsuccessfully', () => {
    username = 'test';
    repoId = 'test_repo';

    repos = [
      {
        name: 'test_repo',
        owner: {
          login: 'username1',
        },
      },
    ];

    const requestUserDescriptor = getReposIterator.next(username).value;
    expect(requestUserDescriptor).toMatchSnapshot();

    const requestRepoDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoDescriptor).toMatchSnapshot();

    const putDescriptor = getReposIterator.next(repos).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USER_NOT_FOUND)),
    );

    const iteration = getReposIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully', () => {
    username = 'test';
    repoId = 'test_repo';

    repos = [
      {
        name: 'test_repo',
        owner: {
          login: 'username1',
        },
        default_branch: 'main',
      },
    ];

    const requestUserDescriptor = getReposIterator.next(username).value;
    expect(requestUserDescriptor).toMatchSnapshot();

    const requestRepoIdDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoIdDescriptor).toMatchSnapshot();

    const requestRepoListDescriptor = getReposIterator.next(repos).value;
    expect(requestRepoListDescriptor).toMatchSnapshot();

    repos = {
      readMe: 'test read me',
      sha: 'test_sha',
      url: 'test_url',
      tree: [],
    };

    const putRepoInfoDescriptor = getReposIterator.next(repos).value;
    expect(putRepoInfoDescriptor).toEqual(
      put(
        slice.githubRepoInfoActions.repoError(RepoErrorType.USER_HAS_NO_REPO),
      ),
    );
  });

  it('should dispatch the reposLoaded action if it requests the repo with tree data successfully', () => {
    username = 'test';
    repoId = 'test_repo';

    repos = [
      {
        name: 'test_repo',
        owner: {
          login: 'username1',
        },
        default_branch: 'main',
      },
    ];

    const requestUserDescriptor = getReposIterator.next(username).value;
    expect(requestUserDescriptor).toMatchSnapshot();

    const requestRepoDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoDescriptor).toMatchSnapshot();

    const requestRepoListDescriptor = getReposIterator.next(repos).value;
    expect(requestRepoListDescriptor).toMatchSnapshot();

    repos = {
      readMe: 'test read me',
      sha: 'test_sha',
      url: 'test_url',
      tree: [
        {
          path: 'README.md',
          mode: '100644',
          type: 'blob',
          sha: '209f8e6e9c3cafa816f61d04eafa6c86552bf3db',
          size: 47,
          url:
            'https://api.github.com/repos/adityapandey/adventofcode/git/blobs/209f8e6e9c3cafa816f61d04eafa6c86552bf3db',
        },
      ],
    };

    const requestRepoInfoDescriptor = getReposIterator.next(repos).value;
    expect(requestRepoInfoDescriptor).toMatchSnapshot();

    const base64Content: any = {
      sha: '209f8e6e9c3cafa816f61d04eafa6c86552bf3db',
      node_id:
        'MDQ6QmxvYjMzMjY2MDg3MDoyMDlmOGU2ZTljM2NhZmE4MTZmNjFkMDRlYWZhNmM4NjU1MmJmM2Ri',
      size: 47,
      url:
        'https://api.github.com/repos/adityapandey/adventofcode/git/blobs/209f8e6e9c3cafa816f61d04eafa6c86552bf3db',
      content:
        'IyBhZHZlbnRvZmNvZGUKQWR2ZW50IG9mIENvZGU6IHNvbHV0aW9ucyBpbiBH\nby4=\n',
      encoding: 'base64',
    };
    const putRepoInfoDescriptor = getReposIterator.next(base64Content).value;

    expect(putRepoInfoDescriptor).toEqual(
      put(slice.githubRepoInfoActions.reposLoaded(repos)),
    );
  });

  it('should dispatch the user not found error', () => {
    username = 'test';
    repoId = 'test';
    repos = [];

    const requestDescriptor = getReposIterator.next(username).value;
    expect(requestDescriptor).toMatchSnapshot();

    const requestRepoIdDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoIdDescriptor).toMatchSnapshot();

    const requestReposDescriptor = getReposIterator.next(repos).value;
    expect(requestReposDescriptor).toMatchSnapshot();

    const putDescriptor = getReposIterator.throw({ response: { status: 404 } })
      .value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USER_NOT_FOUND)),
    );
  });

  it('should dispatch the user has no repo error', () => {
    username = 'test';
    repoId = 'test';
    repos = [];

    const requestDescriptor = getReposIterator.next(username).value;
    expect(requestDescriptor).toMatchSnapshot();

    const requestRepoIdDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoIdDescriptor).toMatchSnapshot();

    const requestReposDescriptor = getReposIterator.next(repos).value;
    expect(requestReposDescriptor).toMatchSnapshot();

    const putDescriptor = getReposIterator.next(repos).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.USER_NOT_FOUND)),
    );
  });

  it('should dispatch the github rate limit error', () => {
    username = 'test';
    repoId = 'test';
    repos = [];

    const requestDescriptor = getReposIterator.next(username).value;
    expect(requestDescriptor).toMatchSnapshot();

    const requestRepoIdDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoIdDescriptor).toMatchSnapshot();

    const requestReposDescriptor = getReposIterator.next(repos).value;
    expect(requestReposDescriptor).toMatchSnapshot();

    const putDescriptor = getReposIterator.throw(new Error('Failed to fetch'))
      .value;
    expect(putDescriptor).toEqual(
      put(
        slice.githubRepoInfoActions.repoError(RepoErrorType.GITHUB_RATE_LIMIT),
      ),
    );
  });

  it('should dispatch the response error', () => {
    username = 'test';
    repoId = 'test';
    repos = [];

    const requestDescriptor = getReposIterator.next(username).value;
    expect(requestDescriptor).toMatchSnapshot();

    const requestRepoIdDescriptor = getReposIterator.next(repoId).value;
    expect(requestRepoIdDescriptor).toMatchSnapshot();

    const requestReposDescriptor = getReposIterator.next(repos).value;
    expect(requestReposDescriptor).toMatchSnapshot();

    const putDescriptor = getReposIterator.throw(new Error('some error')).value;
    expect(putDescriptor).toEqual(
      put(slice.githubRepoInfoActions.repoError(RepoErrorType.RESPONSE_ERROR)),
    );
  });
});

describe('githubRepoInfoSaga Saga', () => {
  const githubRepoFormIterator = githubRepoInfoSaga();
  it('should start task to watch for loadRepos action', () => {
    const takeLatestDescriptor = githubRepoFormIterator.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(slice.githubRepoInfoActions.loadRepoById.type, getRepoById),
    );
  });
});
