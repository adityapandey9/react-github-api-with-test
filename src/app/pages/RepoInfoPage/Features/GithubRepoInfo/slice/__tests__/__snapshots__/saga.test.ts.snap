// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRepoById Saga should dispatch the github rate limit error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the github rate limit error 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the github rate limit error 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the github rate limit error 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the github rate limit error 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/users/test/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the data without default branch unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the data without default branch unsuccessfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the data without default branch unsuccessfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the data without default branch unsuccessfully 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/repos/test/test_repo/git/trees/main?recursive=1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo with tree data successfully 6`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/repos/adityapandey/adventofcode/git/blobs/209f8e6e9c3cafa816f61d04eafa6c86552bf3db",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the reposLoaded action if it requests the repo without tree data unsuccessfully 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/repos/test/test_repo/git/trees/main?recursive=1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the response error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the response error 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the response error 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the response error 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the response error 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/users/test/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the user has no repo error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the user has no repo error 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user has no repo error 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user has no repo error 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user has no repo error 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/users/test/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the user not found error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should dispatch the user not found error 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user not found error 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user not found error 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should dispatch the user not found error 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.github.com/users/test/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should return error if repoId is empty 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should return error if repoId is empty 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should return error if repos is empty 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should return error if repos is empty 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getRepoById Saga should return error if username is empty 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      500,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getRepoById Saga should return error if username is empty 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;
